sse/event_source.go

```diff
@@ -38,23 +38,47 @@ type EventSource struct {
 
 	currentReadCloser *ReadCloser
 	lastEventID       string
-	retryInterval     time.Duration
 	lock              sync.Mutex
 
-	closeOnce *sync.Once
-	closed    chan struct{}
+	closeOnce   *sync.Once
+	closed      chan struct{}
+	retryParams RetryParams
```

Comment:
OK, so we will flatten this out in `EventSource` and add `retryInterval` and `maxRetries` as private member of this struct.